x-pack/plugins/ml/public/application/explorer/explorer.js

```diff
@@ -248,6 +249,10 @@ export class Explorer extends React.Component {
 
     const noJobsFound = selectedJobs === null || selectedJobs.length === 0;
     const hasResults = overallSwimlaneData.points && overallSwimlaneData.points.length > 0;
+    const hasResultsWithAnomalies =
```

Comment:
This check needs amending as it means the 'No anomalies' message is shown in the following situations:

- There are anomalous influencer and records results even though there are no anomalous overall buckets (the current message used below needs improving I think - something like 'No anomalous buckets in selected time range')

![image](https://user-images.githubusercontent.com/7405507/107946740-5a3bc000-6f89-11eb-9666-d767a26d2109.png)

- Changing the anomalies severity threshold to a value higher than highest record score (the current message is probably ok here):

![image](https://user-images.githubusercontent.com/7405507/107947056-c28aa180-6f89-11eb-9201-3ffaa7b5c600.png)

Reply:
Good catch, @peteharverson. Updated `hasResultsWithAnomalies` check in feb00623cf5251b9bf403ba51640d810bf90a519

So `hasResults` means we get something back from the overall bucket endpoint (when there are no anomalous records at all we get empty results from the overall bucket endpoint) and `hasResultsWithAnomalies` means those overall bucket scores are > 0 OR we get some anomaly records back for populating the table.

Reply:
> "....No anomalous buckets in selected time range"

If it's true that this section is related to overall bucket scores, perhaps the message could be something like this:

> No anomalies found in the overall bucket results for this time range

We talk about "overall buckets" in the docs in at least two places, in case it needs to be linked: https://www.elastic.co/guide/en/machine-learning/master/ml-buckets.html and https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-get-overall-buckets.html

Reply:
Thanks for taking a look @lcawl - updated the message in 1e10d7a3019dc185f328e5ae74cb74fe9c0f83e0
Added screen shot to the PR description above as well 👌